### PR TITLE
Support node load and cli publish from a manifest file

### LINKED ISSUE #794: subql-node -f point to a project.yaml
in order to support multiple network

### PR DESCRIPTION
**Pull request by jiqiang90**
Closes #794

### CODE REVIEW

**Comment by ianhe8x** [comment]
Can you add tests for this feature?

**Comment by jiqiang90** [comment]
Need to add SUBQL_ACCESS_TOKEN in $secret

File: packages/cli/src/commands/publish.ts [hunk 1]
```diff
@@ -46,9 +46,7 @@ export default class Publish extends Command {
     }
```
Comment (ianhe8x) [inline packages/cli/src/commands/publish.ts:46]: Seems need to throw here.

File: packages/common/src/project/readers/reader.ts [hunk 2]
```diff
@@ -42,10 +39,10 @@ export class ReaderFactory {
       return new IPFSReader(locationWithoutSchema, options.ipfs);
```
Comment (ianhe8x) [inline packages/common/src/project/readers/reader.ts:39]: logic can be improved here. if `ipfs://` is provided but then is tested to be not valid, shouldn't we throw error?

File: packages/node/src/configure/configure.module.ts [hunk 3]
```diff
@@ -17,6 +17,8 @@ const YargsNameMapping = {
   local: 'localMode',
 };
 
+const IPFS_REGEX = /ipfs:\/\//i;
```
Comment (ianhe8x) [inline packages/node/src/configure/configure.module.ts:20]:
```suggestion
const IPFS_REGEX = /^ipfs:\/\//i;
```

File: packages/node/src/configure/configure.module.ts [hunk 4]
```diff
@@ -36,11 +38,13 @@ function yargsToIConfig(yargs: Args): Partial<IConfig> {
...
 
 function defaultSubqueryName(config: Partial<IConfig>): MinConfig {
+  const ipfsMatch = config.subquery.match(IPFS_REGEX);
   return {
     ...config,
     subqueryName:
-      config.subqueryName ??
-      last(path.resolve(config.subquery).split(path.sep)),
+      config.subqueryName ?? ipfsMatch
+        ? config.subquery.replace('ipfs://', '')
```
Comment (ianhe8x) [inline packages/node/src/configure/configure.module.ts:46]:
```suggestion
        ? config.subquery.replace(IPFS_REGEX, '')
```

File: packages/node/src/configure/configure.module.ts [hunk 5]
```diff
@@ -112,12 +116,17 @@ export class ConfigureModule {
...
-          config.subquery,
-        );
+    let projectPath: string;
+
+    const ipfsMatch = argv.subquery.match(IPFS_REGEX);
+    if (ipfsMatch) {
+      projectPath = argv.subquery;
+    } else {
+      projectPath = path.resolve(
+        config.configDir && !argv.subquery ? config.configDir : '.',
```
Comment (ianhe8x) [inline packages/node/src/configure/configure.module.ts:126]: The origin implementation seems too complicated. argv.subquery should be mandatory then here can be simplified to `projectPath = argv.subquery`, `path.resolve()` can be done in localreader.

**Review by ianhe8x — APPROVED** [review]

**Pull request merged**